test/dummy/app/assets/javascripts/modals.coffee

```diff
@@ -92,6 +92,31 @@ $(document).on 'click', 'a.modal-reload', ->
 		title_icon: 'gear-a'
 	)
 
+## On Show
+
+$(document).on 'click', 'a.modal-onshow', ->
+	closeAction = { title: 'Close', class: 'close-modal' }
+	openNewModal = ->
+		tinyModal.showDirect(
+			"<h1 class=text-center'>Close this modal to execute the callback associated with the previous one!</h1>"
+			title: 'New Modal'
+			actions: [ closeAction ]
+		)
+
+	tinyModal.showDirect(
+		"<h1 class='text-center'>When you open the next modal and close it, a callback associated with this modal will execute and show you an alert!</h1>"
+		title: 'On-show demo'
+		onShow: ->
```

Comment:
I'm not sure `onShow` is the best name for this. Maybe `onPopReveal` or `onReturn` or something? Because it doesn't seem to be called when it is first shown, only on subsequent times.